lambda/putSimDetails.ts

```diff
@@ -1,27 +1,34 @@
 import { type DynamoDBClient, PutItemCommand } from '@aws-sdk/client-dynamodb'
-import { marshall } from '@aws-sdk/util-dynamodb'
 import type { SimDetails } from './getSimDetailsFromCache.js'
 
 export const putSimDetails =
 	(db: DynamoDBClient, cacheTableName: string) =>
-	async (
-		iccid: string,
-		simExisting: boolean,
-		simDetails?: SimDetails,
-		ts?: Date,
-	): Promise<void> => {
+	async ({
+		iccid,
+		simExisting,
+		simDetails,
+		historyTs,
+		ts,
+	}: {
+		iccid: string
+		simExisting: boolean
+		simDetails?: SimDetails
+		historyTs?: Date
+		ts?: Date
+	}): Promise<void> => {
+		const history = historyTs ? { S: historyTs.toISOString() } : { NULL: true }
 		await db.send(
 			new PutItemCommand({
 				TableName: cacheTableName,
-				Item: marshall({
-					iccid,
-					timestamp: new Date().toISOString(),
-					ttl: Date.now() / 1000 + 24 * 60 * 60 * 30, // 30 days
-					usedBytes: simDetails?.usedBytes ?? 0,
-					totalBytes: simDetails?.totalBytes ?? 0,
-					SIMExisting: simExisting,
-					ts: (ts ?? new Date()).toISOString(),
-				}),
+				Item: {
+					iccid: { S: iccid },
+					historyTs: history,
+					ttl: { N: (Date.now() / 1000 + 24 * 60 * 60 * 30).toString() }, // 30 days
+					usedBytes: { N: (simDetails?.usedBytes ?? 0).toString() },
+					totalBytes: { N: (simDetails?.totalBytes ?? 0).toString() },
+					SIMExisting: { BOOL: simExisting },
+					ts: { S: (ts ?? new Date()).toISOString() },
+				},
```

Comment:
Why not use `marshall` here? Saves the manual work to create the DynamoDB item structure.

Reply:
I tried to figure out how to add 'NULL: true' to historyTs and wasn't sure how to do it without specifying NULL. I found an example in our code where marshall was not used for a similar case where the putItemCommand included {NULL: true}.

Reply:
Strange, because `marshall` can deal perfectly fine with null values:

```javascript
import { marshall } from '@aws-sdk/util-dynamodb'
import assert from 'node:assert/strict'

assert.deepEqual(
	marshall({
		foo: null,
	}),
	{ foo: { NULL: true } },
)
```